src/content/scripts/localization.js

```diff
@@ -0,0 +1,72 @@
+/*
+ license: The MIT License, Copyright (c) 2016-2019 YUKI "Piro" Hiroshi
+ original: http://github.com/piroor/webextensions-lib-l10n
+ 
+  Modification by John Bieling:
+   * Auto select ConversionHelper.i18n or browser.i18n
+   * Removed logging
+*/
+
+var localization = {
+  get i18n() {
+    if (!this._i18n) {
+      // Check wether this is called from the WebExtension part of the add-on,
+      // or from the legacy part, where the ConversionHelper provides the API.
+      // If the ConversionHelper is no longer needed, because the add-on has
+      // been converted to a pure WebExtension, this may simply return
+      // browser.1i8n
+      let rv = null;
+      try {
+        rv = ConversionHelper.i18n;
+      } catch (e) {
+        //not there, fallback to browser
+      }
+      
+      this._i18n = rv || chrome.i18n;
+    }
+    
+    return this._i18n;
+  },
+  
+  updateString(string) {
+    return string.replace(/__MSG_(.+?)__/g, matched => {
+      const key = matched.slice(6, -2);
+      return this.i18n.getMessage(key) || matched;
+    });
+  },
+
+  updateSubtree(node) {
+    const texts = document.evaluate(
```

Comment:
document.evaluate (not sure I have used this before, and I am sure it's safe, but might this be forbidden in the future?)

Reply:
This is not an evil eval friend, but an XPATH evaluate:
https://developer.mozilla.org/en-US/docs/Web/API/Document/evaluate

No need to be worried.

Reply:
Aha ok. I looked at XPATH eons ago... can't even remember what I needed it for.